### ISSUE TITLE
[Bug]: Run LLMs with OpenVINO GenAI Flavor on NPU

### ISSUE THREAD

**Issue by taikai-zz** [openvinotoolkit/openvino.genai#1216]
### OpenVINO Version

Name: openvino
Version: 2024.4.0
Summary: OpenVINO(TM) Runtime
Home-page: https://docs.openvino.ai/2023.0/index.html
Author: Intel(R) Corporation
Author-email: openvino@intel.com
License: OSI Approved :: Apache Software License
Location: /root/openvino_env/lib/python3.12/site-packages
Requires: numpy, openvino-telemetry, packaging
Required-by: openvino-tokenizers

### Operating System

Ubuntu 24.04 LTS　　Linux ubuntu 6.8.0-48-generic 

### Device used for inference

NPU

### Framework

None

### Model used

 TinyLlama

### Issue description

Refer to official documentation：
https://docs.openvino.ai/2024/learn-openvino/llm_inference_guide/genai-guide-npu.html

This is my hardware information
![1731632595149](https://github.com/user-attachments/assets/86a62c18-e9f2-48e0-8d1f-14386f8e5ca0)

import openvino_genai as ov_genai
help(ov_genai.LLMPipeline)
![1731632229545](https://github.com/user-attachments/assets/abbb7d16-4a35-4507-ac00-e42669b1630d)
As shown in the above figure: The device does not support NPU. I followed the instructions in the document and changed it to NPU, but the result was empty. Changing it to CPU or GPU restored normal operation. May I ask where I made the operation error?
![1731632595149](https://github.com/user-attachments/assets/0b24c110-7187-4b6b-9aec-52233ff0efba)
![1731633075481](https://github.com/user-attachments/assets/7c27a21c-6b72-4928-b088-55fc0b414312)

There is another issue: if you check the usage rate of NPU in Ubuntu environment, such as tools like nvidia-smi

**Comment by taikai-zz** [comment]
optimum-cli export openvino -m Qwen/Qwen2-7B --weight-format int4 --sym --ratio 1.0 --group-size 128 Qwen2-7B
![1733297624039](https://github.com/user-attachments/assets/d8272fa0-10a1-4675-a8ae-4185b52edfc3)

**Comment by helena-intel** [comment]
Unfortunately, there is an issue with using NPU for LLMs on Ubuntu. The NPU team is working on it; the issue is not with OpenVINO, but on the kernel/driver level. I am sorry you're running into this. We will keep you informed.

**Comment by dmatveev** [comment]
This is 32GB MTL.. Ticket was opened for TinyLLaMa, but logs are mentioning group-quantized QWEN2-7B

**Comment by Wan-Intel** [comment]
I also encountered issue only when using NPU.
<img width="434" alt="npu failed" src="https://github.com/user-attachments/assets/04a8ed8a-edb0-4ca3-93a0-3961eae1b929">

I'll escalate the case to relevant the team and we'll provide an update as soon as possible.

**Comment by helena-intel** [comment]
@taikai-zz a new NPU driver was released today with a fix for LLM on LNL: https://github.com/intel/linux-npu-driver/releases/tag/v1.10.1 Could you check if that fixes the issue for you? We also had a new openvino-genai release this week, 2024.6, with performance improvements on NPU, so please upgrade with `pip install --upgrade openvino-genai`.

Also note that for running larger LLMs (>4B) you should use per-channel quantization. This note will be added to the docs too, I'm mentioning it here because I see you're using a 7B model. Instead of group-size 128, you should specify group-size -1 (note the minus sign). This is an example from the docs for Llama-2-7b: `optimum-cli export openvino -m meta-llama/Llama-2-7b-chat-hf --weight-format int4 --sym --group-size -1 --ratio 1.0 --awq --scale-estimation --dataset=wikitext2  Llama-2-7b-chat-hf`

**Comment by taikai-zz** [comment]
Thank you for your help. It has now returned to normal, but the speed is too slow。
![1735005386604](https://github.com/user-attachments/assets/21db163d-9c49-4e42-84a0-3d06edb17407)

There is an error in the document, please fix it
![1735004820289](https://github.com/user-attachments/assets/8fa66300-3bf7-411a-8c2c-451047c7a2b6)

**Comment by helena-intel** [comment]
I'm glad to hear the issue is fixed! For faster speed, please see the document (same one you screenshotted) about model caching. That will speed up model loading time. Since model loading time only occurs once, it's also useful to measure inference time, by adding `start = time.perf_counter()` and `end = time.perf_counter() before and after `pipe.generate()` and then showing duration as `print(end-start)`. 

The `group_size -1` enables channel-wise quantization, your screenshot is from the group quantization tab. Also note that I recommended this for larger models. For the 1.1B model, group quantization will work fine too. As I mentioned, this will be clarified in the docs.

**Comment by kmaki565** [comment]
I am trying to run the LLM samples with NPU on **Windows**.
The command-line option mentioned above must be `--group-size`, not `--group_size`. With this correction (and I needed `pip install nncf`) I can convert the model, but creating the pipeline on NPU fails (GPU worked well with the same model). 
It looks like the app crashes in `npu_level_zero_umd.dll`. 
I use the latest NPU driver 32.0.100.3104, and confirmed the same crash occurred with MTL and LNL.
![image](https://github.com/user-attachments/assets/7822a34a-7f53-4a56-a7b3-93a02050c2ac)

**Comment by TolyaTalamanov** [comment]
Hi @kmaki565! Could you please clarify the following:
- What model you're trying to run? (HF tag)
- How do you convert model? (optimum-cli command line)
- What is the version for the following components: `nncf`, `onnx`, `optimum-intel`
- What is the version for `openvino`, `openvino-tokenizers`, `openvino-genai`

Thanks!

**Comment by TolyaTalamanov** [comment]
Closing this as the original problem of this issue has been resolved.

@kmaki565 Could you please try out the latest nightly package and file a separate issue if it doesn't work, please?
```
pip install --force-reinstall --pre openvino openvino-tokenizers openvino-genai --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
```

It must be equal or newer than `2025.1.0.0.dev20250306`


Driver: https://www.intel.com/content/www/us/en/download/794734/intel-npu-driver-windows.html

**Comment by kmaki565** [comment]
@TolyaTalamanov Yes, now I can run [the chat sample](https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/python/text_generation/chat_sample.py) in the latest master branch **on NPU**.
Thank you for the update.

**Comment by jinwater88** [comment]
> [@TolyaTalamanov](https://github.com/TolyaTalamanov) Yes, now I can run [the chat sample](https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/python/text_generation/chat_sample.py) in the latest master branch **on NPU**.是的，现在我可以在 NPU 上运行最新主分支中的聊天示例了。 Thank you for the update.感谢您的更新。

I'm currently using a qwen4B int4 for inference on an OpenVino Ultra5 GPU, without using the NPU. Each inference attempt consumes a large amount of RES (Resources Per Memory), leaving no memory available for other models. Are there any optimization solutions?
